sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -168,6 +166,21 @@ class CodegenContext {
   val mutableStates: mutable.ArrayBuffer[(String, String, String)] =
     mutable.ArrayBuffer.empty[(String, String, String)]
 
+  // An array keyed by the tuple of mutable states' types and initialization code, holds the
+  // current max index of the array
+  var mutableStateArrayIdx: mutable.Map[(String, String), Int] =
```

Comment:
From the below code, looks like this is keyed by `(javaType, arrayName)`, instead of `(javaType, initCode)`?

Reply:
Good catch. Old comment still exists.

Reply:
Good catch, old comment still exists there. Thanks